### PR TITLE
[mqtt] Promote client lifecycle logs to info

### PR DESCRIPTION
**Pull request by dmolokanov**
Right now we don't log client lifecycle events when "INFO" is the max log level. It makes it really hard to debug in these circumstances.

`log::info!` when:
- [x] client connected
- [x] client disconnected gracefully
- [x] broker drops connection for the client
- [x] broker refuses to connect a client

### CODE REVIEW

**Review by vadim-kovalyov — CHANGES_REQUESTED** [review]
I'm not sure I agree with these changes. I had quite different experience with the logs when I was bugbashing nested edge + bridge. I think we need to discuss that offline.

**Review by nyanzebra — COMMENTED** [review]
I will approve after addressing Vadim's comment and confirming with Viktor that these log changes are everything that he wants :) Feel free to message me if need the approval

File: mqtt/mqtt-broker/src/broker.rs
Comment (varunpuranik) [inline mqtt/mqtt-broker/src/broker.rs:310]: Is this useful? Do you want to include clientId and reason why it was refused?

File: mqtt/mqtt-broker/src/broker.rs
Comment (varunpuranik) [inline mqtt/mqtt-broker/src/broker.rs:379]: Does this need to be info? Feels like it is only useful when not authorized?

File: mqtt/mqtt-broker/src/broker.rs
Comment (varunpuranik) [inline mqtt/mqtt-broker/src/broker.rs:433]: Again - with multiple clients, I feel it is useful to include the client_id as well.

File: mqtt/mqtt-broker/src/broker.rs
Comment (varunpuranik) [inline mqtt/mqtt-broker/src/broker.rs:997]: client_id?

File: mqtt/mqtt-broker/src/broker.rs
Comment (dmolokanov) [reply mqtt/mqtt-broker/src/broker.rs:310]: client_id is in the context of trace event. also before each refuse_connection call, there is a warning message

Comment (varunpuranik) [reply mqtt/mqtt-broker/src/broker.rs:310]: I am not sure i follow.. 
> client_id is in the context of trace event.
Isn't this change to address the case when only info logging is enabled?
>also before each refuse_connection call, there is a warning message
and that has the info? Why do we still need this then?

Comment (dmolokanov) [reply mqtt/mqtt-broker/src/broker.rs:310]: https://github.com/Azure/iotedge/pull/4525/files#diff-c37c4acce39447d43b2fbcf9c1698aceee26bc6e065afb0f165b85a89f2be923L334-L341

1. before each `refuse_connection!` call we have a warn message explaining what went wrong. so I don't think we need to provide refuse connection message again with info level.
2. since we are using tracing a not logging we can use trace event context and put additional data into it like client_id of the client which packet we process. so we don't need to pass client_id to every log message because it is already there

Comment (vadim-kovalyov) [reply mqtt/mqtt-broker/src/broker.rs:310]: Can you please make this `warn!`

**Review by vadim-kovalyov — COMMENTED** [review]
I'm fine with adding 2 new logs if that helps debugging connectivity issues, but I'm not sure I agree with changing some `debug!` logs to `info!`. For me it is unnecessary and does not help to solve immediate issues.

File: mqtt/mqtt-broker/src/broker.rs
Comment (vadim-kovalyov) [reply mqtt/mqtt-broker/src/broker.rs:310]: Or rather why not update the `debug!` log above? (line 306). Also, it would be good to include reason code.

**Comment by dmolokanov** [comment]
Closing this PR since the whole log should be revised.

**Pull request closed without merging**
